api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/PMQueryBuilder.java

```diff
@@ -0,0 +1,197 @@
+package org.pmiops.workbench.cohortbuilder.querybuilder;
+
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang3.StringUtils;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.pmiops.workbench.model.Attribute;
+import org.pmiops.workbench.model.Operator;
+import org.pmiops.workbench.model.SearchParameter;
+import org.pmiops.workbench.utils.OperatorUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+public class PMQueryBuilder extends AbstractQueryBuilder {
+
+  private static final String BLOOD_PRESSURE = "BP";
+  private static final String HEART_RATE_DETAIL = "HR-DETAIL";
+  private static final String HEIGHT = "HEIGHT";
+  private static final String WEIGHT = "WEIGHT";
+  private static final String BMI = "BMI";
+  private static final String WC = "WC";
+  private static final String HC = "HC";
+  private static final String HR = "HR";
+  private static final String PREG = "PREG";
+  private static final String WHEEL = "WHEEL";
+  private static final String CONCEPT_ID = "conceptId";
+  private static final List<String> PM_TYPES_WITH_ATTR =
+    Arrays.asList(BLOOD_PRESSURE, HEART_RATE_DETAIL, HEIGHT, WEIGHT, BMI, WC, HC);
+  ImmutableMap<String, String> exceptionText = ImmutableMap.<String, String>builder()
+    .put(HEART_RATE_DETAIL, "Heart Rate")
+    .put(HEIGHT, "Height")
+    .put(WEIGHT, "Weight")
+    .put(BMI, "BMI")
+    .put(WC, "Waist Circumference")
+    .put(HC, "Hip Circumference")
+    .put(HR, "Heart Rate")
+    .put(PREG, "Pregnancy")
+    .put(WHEEL, "Wheel Chair User")
+    .build();
+
+  private static final String BP_INNER_SQL_TEMPLATE =
+    "select person_id, measurement_date from `${projectId}.${dataSetId}.measurement`\n" +
```

Comment:
Why does this return measurement_date where the rest don't? How does this get used when selecting participants?